lambda_function.py

```diff
@@ -0,0 +1,629 @@
+    print(
+        json.dumps(
+            {
+                'message': 'Updating record in route53 zone',
+                'response': response
+            },
+            default=str
+        )
+    )
+    print(
+        json.dumps(
+            {
+                'message': 'Record updated',
+                'record': conf['record_name']
+            },
+            default=str
+        )
+    )
+
+def get_item_from_dynamodb():
+    '''Get item from DynamoDB table.'''
+    conf['ID'] = '{}_{}'.format(
+        conf['UserName'],
+        conf['DomainName']
+       #conf['ZoneID'],
+       #conf['Email']
+    )
+    data = {
+        'ID': {
+            'S': conf['ID']
+        }
+    }
+    response = dynamodb.get_item(
+        TableName=conf['dynamodb_table_name'],
+        Key=data
+    )
+   #log.debug('Getting item from DynamoDB - Response: "{}"'.format(response))
+    print(
+        json.dumps(
+            {
+                'message': 'Getting item from DynamoDB',
+                'id': conf['ID'],
+                'key': data
+            },
+            default=str
+        )
+    )
+    print(
+        json.dumps(
+            {
+                'message': 'Getting item from DynamoDB',
+                'response': response
+            },
+            default=str
+        )
+    )
+    conf['dynamodb_record'] = response.get('Item')
+
+def add_item_to_dynamodb():
+    '''Add item to DynamoDB table.'''
+    last_update_time = datetime.datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S')
+    data = {
+        'ID': {
+            'S': conf['ID']
+        },
+        'UserName': {'S': conf['UserName']},
+        'DomainName': {'S': conf['DomainName']},
+        'ZoneID': {'S': conf['ZoneID']},
+        'NameServers': {'S': conf['NameServers']},
+        'Email': {'S': conf['Email']},
+        'LastUpdateUTC': {'S': last_update_time},
+        'SourceIP': {'S': conf['source_ip']}
+    }
+    response = dynamodb.put_item(
+        TableName=conf['dynamodb_table_name'],
+        Item=data
+    )
+   #log.debug('Adding item to DynamoDB - Response: "{}"'.format(response))
+    print(
+        json.dumps(
+            {
+                'message': 'Adding item to DynamoDB',
+                'response': response
+            },
+            default=str
+        )
+    )
+    conf['response'] = 'CREATE OK'
+
+def update_item_in_dynamodb():
+    '''Update item in DynamoDB table.'''
+    last_update_time = datetime.datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S')
+    data = {
+        'ID': {
+            'S': conf['ID']
+        },
+        'UserName': {'S': conf['UserName']},
+        'DomainName': {'S': conf['DomainName']},
+        'ZoneID': {'S': conf['ZoneID']},
+        'NameServers': {'S': conf['NameServers']},
+        'Email': {'S': conf['Email']},
+        'LastUpdateUTC': {'S': last_update_time},
+        'SourceIP': {'S': conf['source_ip']}
+    }
```

Comment:
What is `'S'` ?

Maybe we can simplify structure?
```suggestion
    data = {
        'ID': conf['ID'],
        'UserName': conf['UserName'],
        'DomainName': conf['DomainName'],
        'ZoneID': conf['ZoneID'],
        'NameServers': conf['NameServers'],
        'Email': conf['Email'],
        'LastUpdateUTC': last_update_time,
        'SourceIP': conf['source_ip']
    }
```

Reply:
It is mandatory regarding to documentation of used aws module.